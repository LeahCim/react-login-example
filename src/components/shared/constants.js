export const CREDENTIALS = 'credentials';
export const NO_CREDENTIALS = '';
export const PENDING_CREDENTIALS = null;