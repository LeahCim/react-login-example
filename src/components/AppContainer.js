import React, { Component } from 'react';
import { encode } from 'base-64';

import App from './App';
import { CREDENTIALS, NO_CREDENTIALS, PENDING_CREDENTIALS } from './shared/constants';

export default class AppContainer extends Component {

    state = {
        credentials: PENDING_CREDENTIALS
    }

    componentDidMount() {
        this.loadCredentials();
    }

    loadCredentials = () => {
        this.setState({
            credentials: localStorage.getItem(CREDENTIALS) || NO_CREDENTIALS
        });
    }

    setCredentials = (username, password) => {
        const credentials = encode(`${username}:${password}`);

        this.setState({
            credentials
        });

        localStorage.setItem(CREDENTIALS, credentials);
    }

    resetCredentials = () => {
        this.setState({
            credentials: NO_CREDENTIALS
        });

        localStorage.removeItem(CREDENTIALS);
    }

    render = () =>
        <App
            credentials={this.state.credentials}
            setCredentials={this.setCredentials}
            resetCredentials={this.resetCredentials}
        />
}