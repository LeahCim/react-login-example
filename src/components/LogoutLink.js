import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { LOGIN } from './shared/routes';

export default function LogoutLink({ credentials, resetCredentials }) {
    return !!credentials &&
        <Link to={LOGIN} onClick={resetCredentials}>
            Log out
         </Link>;
}

LogoutLink.propTypes = {
    credentials: PropTypes.string,
    resetCredentials: PropTypes.func.isRequired
}