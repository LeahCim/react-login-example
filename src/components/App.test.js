import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import AppContainer from './AppContainer';
import LogoutLink from './LogoutLink';
import { CREDENTIALS } from './shared/constants';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppContainer />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Shallow renders App', () => {
  const wrapper = shallow(<AppContainer />);
  expect(wrapper).toMatchSnapshot();
});

xit('Passes empty credentials to LogoutLink if session not saved in localStorage', () => {
  const wrapper = shallow(<AppContainer />);
  expect(wrapper.find(LogoutLink)).toHaveProp(CREDENTIALS, "");
});

xit('Passes credentials from localStorage to LogoutLink', () => {
  const credentials = "blah";
  localStorage.setItem(CREDENTIALS, credentials);
  const wrapper = shallow(<AppContainer />);
  expect(wrapper.find(LogoutLink)).toHaveProp(CREDENTIALS, credentials);
});