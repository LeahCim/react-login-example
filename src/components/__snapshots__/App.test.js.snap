// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow renders App 1`] = `
<App
  credentials={null}
  loadCredentials={[Function]}
  resetCredentials={[Function]}
  setCredentials={[Function]}
/>
`;
